src/main/java/me/ryanhamshire/GriefPrevention/PlayerEventHandler.java

```diff
@@ -1658,7 +1659,22 @@ void onPlayerInteract(PlayerInteractEvent event)
 			if(claim != null)
 			{
 				playerData.lastClaim = claim;
-				
+
+				if(clickedBlock.getType() == Material.LECTERN)
+				{
+					String noAccessReason = claim.allowAccess(player);
+					if(noAccessReason != null && instance.config_claims_lecternReadingRequiresAccessTrust)
+					{
+						event.setCancelled(true);
+						GriefPrevention.sendMessage(player, TextMode.Err, noAccessReason);
+						return;
+					}
+					else
+					{
+						return;
+					}
+				}
```

Comment:
I actually prefer the change @zedwick made in #580 - he exempted reading from lecterns from the containertrust check, and added a case to check for them in the accesstrust check. Better than just a special section of code for a single case.

Reply:
You're right, that'll definitely clean up the code quite a bit